src/relay/transforms/to_mixed_precision.cc

```diff
@@ -0,0 +1,409 @@
+    } else {
+      final_category = initial_category;
+    }
+
+    // Create the new arguments to the call.
+    DataType wanted_arg_dtypes =
+        final_category == MIXED_PRECISION_ALWAYS ? mixed_precision_type : DataType::Float(32);
+    auto call_args_and_types = CastAllArgs(post_call_node->args, cur_arg_types, wanted_arg_dtypes);
+    Array<Expr> new_args = call_args_and_types.first;
+    Array<Type> new_arg_types;
+
+    if (pre_call_node->op.as<FunctionNode>()) {
+      // Function Nodes don't store type info in the Call, it should be a []
+      new_arg_types = pre_call_node->type_args;
+    } else {
+      new_arg_types = call_args_and_types.second;
+    }
+
+    // Finally create the new attributes.
+    if (final_category == MIXED_PRECISION_ALWAYS) {
+      Attrs new_attrs = GetNewAttrs(pre_call_node, accumulation_dtype);
+      Expr output = Call(cur_op, new_args, new_attrs, new_arg_types, pre_call_node->span);
+      if (accumulation_dtype != output_dtype) {
+        output = CastArg(output, GetType(output), output_dtype);
+      }
+      return output;
+    }
+
+    return Call(cur_op, new_args, pre_call_node->attrs, new_arg_types, pre_call_node->span);
+  }
+
+  Expr VisitExpr_(const FunctionNode* func) final {
+    // Erase the ret_type annotation and let the normal pass recalculate
+    const_cast<FunctionNode*>(func)->ret_type = Type(nullptr);
+    return ExprMutator::VisitExpr_(func);
+  }
+
+  Expr VisitExpr_(const LetNode* op) final {
+    // First convert as much of the bound computation to lower precision as possible
+    Expr value = this->Mutate(op->value);
+
+    // Then rewrite the var type and associated expression
+    Var var = Downcast<Var>(this->Mutate(op->var));
+    VarNode* mutable_var = const_cast<VarNode*>((op->var).as<VarNode>());
+    mutable_var->type_annotation = GetType(value);
+    mutable_var->checked_type_ = mutable_var->type_annotation;
+
+    // Mutate body last as it may depend on previous results
+    Expr body = this->Mutate(op->body);
+    return Let(var, value, body, op->span);
+  }
+};
+
+Expr ToMixedPrecision(const Expr& expr, const DataType& mixed_precision_type,
+                      bool ignore_missing_ops, bool warn_missing_ops) {
+  MixedPrecisionPass converter =
+      MixedPrecisionPass(mixed_precision_type, ignore_missing_ops, warn_missing_ops);
+  auto result = converter.Mutate(expr);
+  return result;
+}
+
+namespace transform {
+
+Pass ToMixedPrecision(DataType mixed_precision_type, bool ignore_missing_ops,
+                      bool warn_missing_ops) {
+  runtime::TypedPackedFunc<Function(Function, IRModule, PassContext)> pass_func =
+      [=](Function f, IRModule m, PassContext pc) {
+        return Downcast<Function>(
+            ToMixedPrecision(f, mixed_precision_type, ignore_missing_ops, warn_missing_ops));
+      };
+  return CreateFunctionPass(pass_func, 10, "ToMixedPrecision", {});
```

Comment:
Is 10 the right level? It should be 0 IMHO, as this pass is supposed to be used by user manually instead of being managed by the pass manager.

Reply:
Not sure, I based this on the level for the fake_quantization_to_integer pass @mbrookhart wrote. I'm not too sure with the guideline behind this so any reference material would be welcome.

I will change it to 0 if there are no objections.

Reply:
Done